sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -114,6 +115,16 @@ class IncrementalExecution(
           stateInfo = Some(nextStatefulOperationStateInfo),
           batchTimestampMs = Some(offsetSeqMetadata.batchTimestampMs),
           eventTimeWatermark = Some(offsetSeqMetadata.batchWatermarkMs))
+
+      case j @ StreamingSymmetricHashJoinExec(lKeys, rKeys, _, cond, _, _, _, left, right) =>
+        j.copy(
+          stateInfo = Some(nextStatefulOperationStateInfo),
```

Comment:
So, I think this may not be robust. At each trigger, we create a new IncrementalExecution, so the `statefulOperatorId` we hope gets incremented in a deterministic manner.
I can imagine adding things to the Optimizer in the future which may move an `EquiJoin` before an aggregation.
In this case, the state store id's of the aggregation and join may switch. I'm not sure if we're protected against that somehow, so just wanted to bring it up.

Reply:
Whatever optimization takes place, the same optimizations will occur in EVERY batch. So if aggregation is pushed below join, then all the batches will have that. 

What we have to guard against is cost-based optimization that can reorder things differently in different batches. That is, why I have disabled cost-based join optimization. And when adding such optimizations in the future, we have to be cautious for the streaming case and disable them.

Also, this is a general concern with other stateful ops as well, not something that this PR would address.

Reply:
That is the case **if** you don't change Spark versions. A more recent Spark version may include new optimizer rules that may change the ordering.
Just something to think about. Would be nice to add a test with aggregation + join and join + aggregation.